src/utils/eigen_utils.hpp

```diff
@@ -0,0 +1,16 @@
+#include <Eigen/Dense>
+#include <vector>
+
+namespace bayesmix {
+// Stacks a list of matrices by row, i.e. concatenating them one
+// on top of the other
+// TODO test
+Eigen::MatrixXd vstack(const std::vector<Eigen::MatrixXd> &mats);
+
+// TODO test
+void append_by_row(Eigen::MatrixXd *a, const Eigen::MatrixXd &b);
+
+Eigen::MatrixXd append_by_row(const Eigen::MatrixXd &a,
+                              const Eigen::MatrixXd &b);
```

Comment:
I feel like these two functions should have different names, since they have different behaviours, one with assignment and one in-place. Or maybe be the same function with an in-place flag. Or maybe I have been spoiled by Python...

Reply:
I'm not really sure I agree, after all c++ overloading is meant for situations like these.
Still if you have a strong opinion I can change that

Reply:
Whatever, no strong feelings